llmfoundry/data/finetuning/collator.py

```diff
@@ -320,24 +323,29 @@ def _process_and_batch_decoder_only(
     ) -> Dict[str, torch.Tensor]:
         # Steps explained in comments
         processed_examples = []
-        for example in examples:
-            input_ids, labels = stitch_turns_decoder_only(
-                example_turns=example['turns'],
-                target_prompts=self.target_prompts,
-                target_responses=self.target_responses,
-                eos_token_id=self.tokenizer.eos_token_id,
-            )
+        input_ids_and_labels = [stitch_turns_decoder_only(
+            example_turns=example['turns'],
+            target_prompts=self.target_prompts,
+            target_responses=self.target_responses,
+            eos_token_id=self.tokenizer.eos_token_id,
+        ) for example in examples]
+
+        if self._is_profiling:
+            max_seq_len = max([len(input_ids) for input_ids, _ in input_ids_and_labels])
```

Comment:
Is this doing almost the same thing as `allow_pad_trimming`? @alextrott16, do you have any thoughts here on how/if we can consolidate this functionality? We're avoiding padding to the max seq length because it's too slow at super long seq length and with packing, we trim the padding off anyway.

Reply:
As discussed offline, yeah this has _nearly_ the same behavior as `allow_pad_trimming` but the latter doesn't trim the first batch. And that's a problem during profiling since it just grabs a single batch.

So, I recommend sticking with the way @b-chu has it implemented here :)